File: lobwebapp-core/src/main/java/com/espindola/lobwebapp/service/contract/ProductService.java
package com.espindola.lobwebapp.service.contract;

import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;

import com.espindola.lobwebapp.domain.Product;
import com.espindola.lobwebapp.exception.NotFoundException;
import com.espindola.lobwebapp.service.contract.base.EntityService;

public interface ProductService extends EntityService<Product> {
	
	Product findByName(String name) throws NotFoundException;
	
	Page<Product> findAllByNameLike(String name, Pageable pageable);

}

File: lobwebapp-client/src/app/service/contract/ProductService.ts
///<reference path="../../reference.ts"/>

module service.contract {
    export interface ProductService extends base.EntityService<domain.Product> {

        findByName(name: string,
            successCallback: (data: domain.Product[], status: number, headers: (headerName: string) => string, config: ng.IRequestConfig) => any,
            errorCallback: (data: domain.util.MessageResponse, status: number, headers: (headerName: string) => string, config: ng.IRequestConfig) => any,
            pageable: domain.util.Page): void;

        listCategory(
            successCallback: (data: string[], status: number, headers: (headerName: string) => string, config: ng.IRequestConfig) => any,
            errorCallback: (data: domain.util.MessageResponse, status: number, headers: (headerName: string) => string, config: ng.IRequestConfig) => any
            ): void;
        
        getImageUrl(productId: number): string;
        getMarkUp(product: domain.Product): number;
    }
}